### DISCUSSION TITLE
Prepping and making a BLAST DB

### DISCUSSION THREAD

**Discussion by benfulton** [bbuchfink/diamond#640] [Q&A] [answered]
We've got an NCBI protein database set up, so it's in the form nr.00.ppi, nr.00.psq, nr.01.psq, nr.01.psq, etc. up to nr.69.psq. To use diamond, we've prepped the database with the command

`diamond prepdb -d nr
`

Looks like this creates a set of ACC files, one for each nr index. But now, how do I make a database using these files? I tried

`diamond makedb --in nr --db nr.dmnd --taxonmap prot.accession2taxid.FULL.gz --taxonnodes nodes.dmp --taxonnames names.dmp`

but it thinks that nr is the name of a file here.

**Comment by bbuchfink** [comment]
`makedb` is for building a database from a fasta file. If you use `prepdb` on a blast db you can then directly use it with diamond, without running `makedb`.

**Comment by benfulton** [comment]
How do I get this to work? I tried

```
$ echo $BLASTDB
/N/soft/blastdb

$ ls $BLASTDB/nr.00*
/N/soft/blastdb/nr.00.acc  /N/soft/blastdb/nr.00.pog	 /N/soft/blastdb/nr.00.tar.gz.extracted
/N/soft/blastdb/nr.00.phd  /N/soft/blastdb/nr.00.ppd	 /N/soft/blastdb/nr.00.tar.gz.md5
/N/soft/blastdb/nr.00.phi  /N/soft/blastdb/nr.00.ppi	 /N/soft/blastdb/nr.00.tar.gz.md5receipt
/N/soft/blastdb/nr.00.phr  /N/soft/blastdb/nr.00.psq
/N/soft/blastdb/nr.00.pin  /N/soft/blastdb/nr.00.tar.gz

$ diamond blastp --in SBFOOD.faa -d nr -e 1e-12 --more-sensitive --threads 24 -f 6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle staxids sskingdoms -o test_diamond
diamond v2.0.13.151 (C) Max Planck Society for the Advancement of Science
Documentation, support and updates available at http://www.diamondsearch.org
Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)

#CPU threads: 24
Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)
Temporary directory: 
#Target sequences to report alignments for: 25
Opening the database... No such file or directory
 [0.001s]
Error: Error calling stat on file nr

```

**Reply by bbuchfink** [reply, marked as answer]
Diamond does not read the `BLASTDB` environment variable. You have to specify the full path to the database.
